provider/gateway/rest/router.go

```diff
@@ -178,6 +183,86 @@ func newJwtServerRouter(addr sdk.Address, privateKey interface{}, jwtExpiresAfte
 	return router
 }
 
+func newResourceServerRouter(log log.Logger, providerAddr sdk.Address, publicKey *ecdsa.PublicKey, lokiGwAddr string) *mux.Router {
+	router := mux.NewRouter()
+
+	// add a middleware to verify the JWT provided in Authorization header
+	router.Use(func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			// verify the provided JWT
+			token, err := jwt.ParseWithClaims(r.Header.Get("Authorization"), &ClientCustomClaims{}, func(token *jwt.Token) (interface{}, error) {
+				// return the public key to be used for JWT verification
+				return publicKey, nil
+			})
+			if err != nil {
+				log.Error(err.Error())
+				http.Error(w, err.Error(), http.StatusUnauthorized)
+				return
+			}
+			// delete the Authorization header as it is no more needed
+			r.Header.Del("Authorization")
+
+			// store the owner & provider address in request context to be used in later handlers
+			ownerAddress, err := sdk.AccAddressFromBech32(token.Claims.(*ClientCustomClaims).Subject)
+			if err != nil {
+				log.Error(err.Error())
```

Comment:
If possible just log something simple in cases like this 

```
log.Error("failed parsing owner address", "error", err, "address", address)
```

So that this is easier to debug. Usually the way we'll encounter this is someone like @sacreman reporting the error in the logs and we might have to figure out what is going on.

Reply:
Done